crates/factor-key-value/tests/factor_test.rs

```diff
@@ -224,7 +227,16 @@ async fn multiple_custom_key_value_uses_first_store() -> anyhow::Result<()> {
     Ok(())
 }
 
-struct TomlConfig(Option<toml::Table>);
+struct TomlConfig {
+    resolver: Arc<RuntimeConfigResolver>,
```

Comment:
This is quite awkward; I think we can probably do better by making the `Factor` available in `get_runtime_config`.

Reply:
Lann and I talked about this, and came to the tentative conclusion that while having shared state between the factor and the runtime config source is maybe not 100% ideal, it works well in this case. We'll continue to keep an eye on this area and continue to adjust as we get more comfortable with the consequences of the current design.